packages/dd-trace/src/remote_config/index.js

```diff
@@ -0,0 +1,260 @@
+'use strict'
+
+const uuid = require('crypto-randomuuid')
+const { EventEmitter } = require('events')
+const Scheduler = require('./scheduler')
+const tracerVersion = require('../../../../package.json').version
+const request = require('../exporters/common/request')
+const log = require('../log')
+
+const clientId = uuid()
+
+const POLL_INTERVAL = 5e3
+
+// There MUST NOT exist separate instances of RC clients in a tracer making separate ClientGetConfigsRequest
+// with their own separated Client.ClientState.
+class RemoteConfigManager extends EventEmitter {
+  constructor (config) {
+    super()
+
+    this.scheduler = new Scheduler((cb) => this.poll(cb), POLL_INTERVAL)
+
+    this.requestOptions = {
+      method: 'POST',
+      url: config.url,
+      hostname: config.hostname,
+      port: config.port,
+      // TODO: do we need hostname/port here ?
+      path: '/v0.7/config'
+    }
+
+    this.state = {
```

Comment:
Is this format documented publicly anywhere? At worst the agent code would do, but we should probably have a reference somewhere to know what should be included, especially as things are added.

Reply:
There is the protobuff definition file [here](https://github.com/DataDog/datadog-agent/blob/main/pkg/proto/datadog/remoteconfig/remoteconfig.proto), but it doesn't have explanations on the fields.

Reply:
Better than nothing, at least it documents the available fields.